cvat-ui/src/actions/import-actions.ts

```diff
@@ -100,40 +84,53 @@ export const importDatasetAsync = (
         const instanceType = getInstanceType(instance);
         const resource = instanceType === 'project' ? 'dataset' : 'annotation';
 
+        const listenForImport = (rqID: string) => core.requests.listen(rqID, {
+            callback: (updatedRequest) => updateRequestProgress(updatedRequest, dispatch),
+        });
+
         try {
             const state: CombinedState = getState();
 
             if (instanceType === 'project') {
                 dispatch(importActions.importDataset(instance, format));
-                const rqID = await (instance as Project).annotations
-                    .importDataset(format, useDefaultSettings, sourceStorage, file, {
+                const rqID = await (instance as Project).annotations.importDataset(
+                    format,
+                    useDefaultSettings,
+                    sourceStorage,
+                    file,
+                    {
                         convMaskToPoly,
                         updateStatusCallback: (message: string, progress: number) => (
                             dispatch(importActions.importDatasetUpdateStatus(
                                 instance, Math.floor(progress * 100), message,
                             ))
                         ),
-                    });
-                if (shouldListenForProgress(rqID, state.requests)) {
-                    await listen(rqID, dispatch);
-                }
+                    },
+                );
+
+                await listenForImport(rqID);
             } else if (instanceType === 'task') {
                 dispatch(importActions.importDataset(instance, format));
-                const rqID = await (instance as Task).annotations
-                    .upload(format, useDefaultSettings, sourceStorage, file, {
-                        convMaskToPoly,
-                    });
-                if (shouldListenForProgress(rqID, state.requests)) {
-                    await listen(rqID, dispatch);
-                }
+                const rqID = await (instance as Task).annotations.upload(
+                    format,
+                    useDefaultSettings,
+                    sourceStorage,
+                    file,
+                    { convMaskToPoly },
+                );
+                await listenForImport(rqID);
             } else { // job
                 dispatch(importActions.importDataset(instance, format));
-                const rqID = await (instance as Job).annotations
-                    .upload(format, useDefaultSettings, sourceStorage, file, {
-                        convMaskToPoly,
-                    });
+                const rqID = await (instance as Job).annotations.upload(
+                    format,
+                    useDefaultSettings,
+                    sourceStorage,
+                    file,
+                    { convMaskToPoly },
+                );
+
                 if (shouldListenForProgress(rqID, state.requests)) {
```

Comment:
Let's remove the `shouldListenForProgress` function here, just as we did in `export-actions`. Since it’s now only used in one place, we can delete it entirely.